pkg/runtime/grpc/grpc-runtime.go

```diff
@@ -110,35 +138,50 @@ func (r *Runtime) Close() error {
 }
 
 func (r *Runtime) ParamDescs() params.ParamDescs {
-	return params.ParamDescs{
-		{
-			Key:         ParamNode,
-			Description: "Comma-separated list of nodes to run the gadget on",
-			Validator: func(value string) error {
-				nodes := strings.Split(value, ",")
-				nodeMap := make(map[string]struct{})
-				for _, node := range nodes {
-					if _, ok := nodeMap[node]; ok {
-						return fmt.Errorf("duplicated node: %s", node)
+	p := params.ParamDescs{}
+	switch r.connectionMode {
+	case ConnectionModeDirect:
+		p.Add(params.ParamDescs{
+			{
+				Key:          ParamRemoteAddress,
+				Description:  "Comma-separated list of remote address (gRPC) to connect to",
+				DefaultValue: api.DefaultDaemonPath,
+			},
+		}...)
+		return p
```

Comment:
nit: how about just returning the slice here without having to use `p.Add()`?

Reply:
When we add the persistency/long running gadgets, there will be a couple of shared parameters that p is initialized with. I wanted to keep the changes minimal, so I didn't clean this up just to move it again with the next PR. Maybe I should add notes/TODOs in such cases.

Reply:
Fair enough.